docs/core/logger.md

```diff
@@ -207,6 +207,74 @@ When debugging in non-production environments, you can log the incoming event us
 
 Use `POWERTOOLS_LOGGER_LOG_EVENT` environment variable to enable or disable (`true`/`false`) this feature. When using Middy.js middleware or class method decorator, the `logEvent` option will take precedence over the environment variable.
 
+### Setting a Correlation ID
+
+To gest started, install `@aws-lambda-powertools/jmespath` package and configure the search function for the correlationId:
+
+=== "Setup the Logger to use JMESPath search"
+
+    ```typescript
+    --8<-- "examples/snippets/logger/correlationIdLogger.ts"
+    ```
+
+???+ tip
+    You can retrieve correlation IDs via `getCorrelationId` method.
+
+You can set a correlation ID using `correlationIdPath` parameter by passing a JMESPath expression, including our custom JMESPath functions or set it manually by calling `setCorrelationId` function.
+
+=== "Setting correlation ID manually"
+
+    ```typescript
+    --8<-- "examples/snippets/logger/correlationIdManual.ts"
+    ```
```

Comment:
```suggestion
    ```

    1. Alternatively, if the payload is more complex you can use a JMESPath expression as second parameter when prividing a search function in the constructor.
```